packages/multichain-network-controller/src/MultichainNetworkController.ts

```diff
@@ -0,0 +1,195 @@
+import type { AccountsControllerSetSelectedAccountAction } from '@metamask/accounts-controller';
+import {
+  BaseController,
+  type ControllerGetStateAction,
+  type ControllerStateChangeEvent,
+  type RestrictedControllerMessenger,
+} from '@metamask/base-controller';
+import type {
+  NetworkStatus,
+  NetworkControllerGetNetworkConfigurationByNetworkClientId,
+  NetworkControllerSetActiveNetworkAction,
+  NetworkControllerGetStateAction,
+  NetworkControllerStateChangeEvent,
+} from '@metamask/network-controller';
+import type { Draft } from 'immer';
+
+import {
+  bitcoinCaip2ChainId,
+} from './constants';
+
+const controllerName = 'MultichainNetworkController';
+
+export type MultichainNetworkMetadata = {
+  features: string[];
+  status: NetworkStatus;
+};
+
+export type MultichainNetworkConfiguration = {
+  chainId: string; // Should be Caip2 type
+  name: string;
+  nativeCurrency: string; // Should be Caip19 type
+  blockExplorerUrls: string[];
+  defaultBlockExplorerUrlIndex?: number;
+  lastUpdated?: number;
+  isEvm?: false;
+};
+
+/**
+ * State used by the {@link MultichainNetworkController} to cache network configurations.
+ */
+export type MultichainNetworkControllerState = {
+  multichainNetworkConfigurationsByChainId: Record<
+    string,
+    MultichainNetworkConfiguration
+  >;
+  selectedMultichainNetworkChainId: string;
+  multichainNetworksMetadata: Record<string, MultichainNetworkMetadata>;
+  nonEvmSelected: boolean;
```

Comment:
What are your thoughts on adding JSDoc for these properties? In particular it's not clear to me what this property does.

Reply:
Also, instead of "storing" a state like this, we could compute differently maybe.

**Unless we really need this to be in the state?**

If we assume that `selectedMultichainNetworkChainId` will also be set to a CAIP-2 chain ID for EVM, we could compute it like:

```ts
isNonEvmSelectedNetwork(): boolean {
   const { namespace } = parseCaipChainId(this.state.selectedMultichainNetworkChainId);
   
   // 'eip155' is the CAIP-2 namespace for EVMs.
   return namespace !== KnownCaipNamespace.Eip155;
}
```

IMO, that's "safer" to compute it dynamically to avoid any de-sync logic between the 2, WDYT?

> [!NOTE]
> This is true only if `selectedMultichainNetworkChainId` also references EVM chain IDs.

Reply:
Hey both, thanks for the review!
Hmmm thats interesting, so instead of this nonEvmSelected, we would have the "real" selected chain Id duplicated in a state variable  `selectedMultichainNetworkChainId`



I dont believe `nonEvmSelected` it trully needs to be in the state! But I see it easier to compute just call a util function, but I don't have any fundamentals to this affirmationa, just seem less effort for the compiler!

I do see that we would still need to have in sync two variables, (the evm or non evm selected chain and the selectedMultichainNetworkChainId) unless I understood your suggestion completely wrong 🙏

@Cal-L you were suggesting a similar approach yesterday, any thoughts?

Reply:
Ahhh, yes we do need it! because It's used to know if it's an evm network selected or a non evm network selected! 
I think the JSDoc will help to understand!